proxy/proxy.go

```diff
@@ -304,6 +307,10 @@ func containerShouldAttach(container *docker.Container) bool {
 	return len(container.Config.Entrypoint) > 0 && container.Config.Entrypoint[0] == weaveWaitEntrypoint[0]
 }
 
+func containerIsWeaveRouter(container *docker.Container) bool {
+	return strings.HasPrefix(container.Config.Image, "weaveworks/weave")
```

Comment:
This prevents users from fetching their images from a different repo, which is something the weave script permits by setting `DOCKERHUB_USER`.